### ISSUE TITLE
[Bug-Candidate]: Echidna takes too long to exit

### ISSUE THREAD

**Issue by aviggiano** [crytic/echidna#1124]
### Describe the issue:

I am running Echidna on a relatively small codebase, but it is taking too long to exit (around 45 seconds)

Initially, I thought it was due to the time spent saving the corpus, but after some measurements (`watch du echidna`, to check for the `corpusDir` size), I believe that's not the case:

https://github.com/crytic/echidna/assets/3029017/b6a4117e-0619-46b4-b992-cd854f6df036

Seems like it's another cleanup process taking too long.
Also, if I Ctrl+C, even after saving the corpus, the HTML will be empty.

### Code example to reproduce the issue:

N/A

### Version:

Echidna 2.2.1
slither 0.9.3 PR https://github.com/crytic/slither/pull/2064

### Relevant log output:

```shell
N/A
```

**Comment by ggrieco-tob** [comment]
I think we are duplicating the saving of the corpus, since all the entries should be saved on the fly. What do you think @samalws-tob ?

**Comment by samalws-tob** [comment]
Yeah I think that's true, although it shouldn't be too much of an issue since corpus saving doesn't take up a lot of time in the video

**Comment by elopez** [comment]
Is this still an issue @aviggiano ? I think we did some improvements in this front a while back

**Comment by aviggiano** [comment]
I haven't experienced this in a long time

We can close this issue
